packages/esm-patient-vitals-app/src/common/helpers.ts

```diff
@@ -33,24 +33,41 @@ export function assessValue(value: number | undefined, range?: ObsReferenceRange
   return 'normal';
 }
 
+export function interpretFhirInterpretation(interpretation: FHIRInterpretation): ObservationInterpretation {
+  switch (interpretation) {
+    case 'Critically Low':
+      return 'critically_low';
+    case 'Critically High':
+      return 'critically_high';
+    case 'High':
+      return 'high';
+    case 'Low':
+      return 'low';
+    case 'Normal':
+      return 'normal';
+  }
+}
+
 export function interpretBloodPressure(
   systolic: number | undefined,
   diastolic: number | undefined,
   concepts: { systolicBloodPressureUuid?: string; diastolicBloodPressureUuid?: string } | undefined,
   conceptMetadata: Array<ConceptMetadata> | undefined,
+  systolicInterpretation?: ObservationInterpretation,
+  diastolicInterpretation?: ObservationInterpretation,
 ): ObservationInterpretation {
   if (!conceptMetadata) {
     return 'normal';
   }
 
-  const systolicAssessment = assessValue(
-    systolic,
-    getReferenceRangesForConcept(concepts?.systolicBloodPressureUuid, conceptMetadata),
-  );
+  const systolicAssessment =
+    systolicInterpretation ??
+    assessValue(systolic, getReferenceRangesForConcept(concepts?.systolicBloodPressureUuid, conceptMetadata));
 
-  const diastolicAssessment = concepts?.diastolicBloodPressureUuid
-    ? assessValue(diastolic, getReferenceRangesForConcept(concepts.diastolicBloodPressureUuid, conceptMetadata))
-    : 'normal';
+  const diastolicAssessment =
+    diastolicInterpretation ?? concepts?.diastolicBloodPressureUuid
+      ? assessValue(diastolic, getReferenceRangesForConcept(concepts.diastolicBloodPressureUuid, conceptMetadata))
+      : 'normal';
```

Comment:
```suggestion
    diastolicInterpretation ??
    (concepts?.diastolicBloodPressureUuid
      ? assessValue(diastolic, getReferenceRangesForConcept(concepts.diastolicBloodPressureUuid, conceptMetadata))
      : 'normal');
```

There's an operator precedence issue with the current code. It evaluates as `(diastolicInterpretation ?? concepts?.diastolicBloodPressureUuid) ? ... : 'normal'`, which means:
 - If `diastolicInterpretation` is `undefined` and `concepts?.diastolicBloodPressureUuid` exists, it would use the UUID string as a truthy value.
- This incorrectly skips the `assessValue` call when an interpretation should be used

In the suggested code, we add parentheses to correctly check `diastolicInterpretation` first, then fall back to the conditional assessment.